lib/stockboy/candidate_record.rb

```diff
@@ -82,6 +83,21 @@ def partition(filters={})
       nil
     end
 
+    # Generate an md5 hash from the mapped output attributes
+    #
+    # @return [String] MD5 hash
+    #
+    def hash
+      Digest::MD5.hexdigest(Marshal::dump(to_hash)).freeze
```

Comment:
On second thought, let's call this out in the readme, since doing unstable things in the translation should be a no-no, and you should expect this to change.

Reply:
On another note, I'm interested to the reason for the `.freeze`? If a lot of hashes are generated does this not fill up the heap in ruby, as I read frozen objects aren't GC'd?

Reply:
Frozen objects are GC'd the same as unfrozen objects: when they are no longer referenced. The main difference with Strings is that `.freeze` has an optimization to reference the same string object instance if it already exists (since it is not mutable, a comparison on its value is possible).

It's not technically necessary to freeze this, but a string attribute such as this shouldn't be considered "mutable" so it's just good form to freeze it.

e.g. `record.hash << "-mutant"` might look like it is changing an attribute in-place. Raising an error lets you know that this is not expected. Also, if this attribute was cached on the instance, we wouldn't want to allow mutating it.